### ISSUE TITLE
Pylance can not find parameters in sklearn classes

### ISSUE THREAD

**Issue by ss52** [microsoft/pylance-release#1139]
## Environment data

-   Language Server version: 2021.4.0
-   OS and version: Win 10 (10.0.21337 build 21337)
-   Python version: 3.8.8, scikit-learn 0.24.1

## Actual behaviour

I get a warning from pylance - "No parameter named "feature_range" for a class MinMaxScaler. But this is valid parameter and code works well. Also I have such problems with all parameters from scikit-learn classes.

## Code Snippet / Additional information

```python
from sklearn.preprocessing import MinMaxScaler
s1 = MinMaxScaler(feature_range=(0, 1))
```

**Comment by bschnurr** [comment]
Seems that version of sklearn has a decorator around the `MinMaxScaler.__init__ `that is hiding the signature. 
@_deprecate_positional_args

Best to log an issue with scitkit-learn.
along with some tips on how to write a stub for a decorator here. https://github.com/microsoft/pyright/blob/master/docs/type-stubs.md#cleaning-up-generated-type-stubs

**Comment by bschnurr** [comment]
for now adding this above `def _deprecate_positional_args `

in sklearn/utils/validation.py fixes it
```
from typing import Any, Callable, TypeVar, Optional, overload


_T = TypeVar('_T', bound=Callable[..., Any])

@overload
def _deprecate_positional_args(func: _T) -> _T: ...

@overload
def _deprecate_positional_args(func:None, *, version="1.0 (renaming of 0.25)") -> Callable[[_T], _T]:...

```

thanks @erictraut

**Comment by jakebailey** [comment]
The next release includes a partial stub for this module, which should fix these issues.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.4.2, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202142-21-april-2021
